### PR TITLE
[FEATURE] Delete profilers from DataContext

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- `delete_profiler` method and related tests

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/rule_based_profiler/test_toolkit.py [hunk 1]
```diff
@@ -47,6 +67,50 @@ def test_get_profiler_non_existent_profiler_raises_error(
...
+        delete_profiler(
+            profiler_store=populated_profiler_store,
+            name="my_profiler",
+            ge_cloud_id=None,
+        )
+
+    assert mock_remove_key.call_count == 1
+    assert mock_remove_key.call_args == mock.call(
+        key=ConfigurationIdentifier("my_profiler")
+    )
```
Comment (cdkini) [inline tests/rule_based_profiler/test_toolkit.py:85]: This feature inherently requires mocks due to the `ProfilerStore`'s reliance on the filesystem.

In order to keep this a true unit test, let's check that the proper key is provided to `remove_key`. The store's backend is already under test so this is simply asserting that we're able to invoke that method with the appropriate arguments.

File: tests/rule_based_profiler/test_toolkit.py [hunk 2]
```diff
@@ -9,7 +9,27 @@
...
-from great_expectations.rule_based_profiler.toolkit import get_profiler, list_profilers
+from great_expectations.rule_based_profiler.toolkit import (
+    delete_profiler,
+    get_profiler,
+    list_profilers,
+)
+
+
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+def test_get_profiler_with_too_many_args_raises_error(
```
Comment (cdkini) [inline tests/rule_based_profiler/test_toolkit.py:20]: Adding this to the prior `get` method to ensure parity.

File: great_expectations/rule_based_profiler/toolkit.py [hunk 3]
```diff
@@ -20,6 +20,10 @@ def get_profiler(
     name: Optional[str] = None,
     ge_cloud_id: Optional[str] = None,
 ) -> RuleBasedProfiler:
+    assert bool(name) ^ bool(
+        ge_cloud_id
+    ), "Must provide either name or ge_cloud_id (but not both)"
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/toolkit.py:25]: ooo fancy :) love it.

File: great_expectations/rule_based_profiler/toolkit.py [hunk 4]
```diff
@@ -53,6 +57,30 @@ def get_profiler(
...
+    else:
+        key = ConfigurationIdentifier(configuration_key=name)
+
+    try:
+        profiler_store.remove_key(key=key)
+    except (ge_exceptions.InvalidKeyError, KeyError) as exc_ik:
+        id_ = key.configuration_key if isinstance(key, ConfigurationIdentifier) else key
+        raise ge_exceptions.ProfilerNotFoundError(
+            message=f'Non-existent Profiler configuration named "{id_}".\n\nDetails: {exc_ik}'
+        )
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/toolkit.py:81]: really like this error message too

File: tests/rule_based_profiler/test_toolkit.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/rule_based_profiler/test_toolkit.py:85]: Ok this explanation makes a lot of sense :) Thank's for the thought you put into this distinction @cdkini

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you sir @cdkini, as always

**Pull request merged**